quote_manager_cli/database.py

```
@@ -0,0 +1,100 @@
+import os
+import sys
+from datetime import datetime
+from typing import Any, Optional, Type
+
+from dotenv import load_dotenv
+from sqlalchemy import Column, DateTime, Integer, Sequence, String, create_engine, inspect
+from sqlalchemy.orm import Session, declarative_base, sessionmaker
+
+from .logger_config import error_logger, info_logger
+
+load_dotenv()
+
+# Fetch database path from environment variable, with a default fallback
+DATABASE_FILE = os.getenv("DATABASE_PATH", "default.db")
+DATABASE_URL = f"duckdb:///{DATABASE_FILE}"
+
+Base: Type[Any] = declarative_base()
+
+
+# Define the Quote model
+class Quote(Base):
+    __tablename__ = "quotes"
+
+    id = Column(Integer, Sequence("id"), primary_key=True)
+    text = Column(String, index=True)
+    author = Column(String)
+    category = Column(String)
+    created_at = Column(DateTime, default=datetime.now)
+
+
+def get_engine(url: str = DATABASE_URL) -> Any:
+    """Creates and returns a new database engine."""
+    try:
+        engine = create_engine(url)
+        return engine
+    except Exception as e:
+        error_logger.error(f"Error creating engine: {e}", exc_info=True)
+    return None
+
+
+def create_session(engine: Any) -> Optional[Session]:
```

Session isn't the only return type for this function. Also why not try to use `Union` in place of `Optional`? Apply this across all files

---

Added a type for engine. I found that `Optional` works as Union[Session, None] would work. But when I used it, I still had errors. to fix that, I just raised the errors instead of returning None.